src/SonarScanner.MSBuild.Shim/PathHelper.cs

```diff
@@ -78,5 +87,28 @@ public static string[] GetParts(DirectoryInfo directory)
             parts.Add(directory.Name);
             return parts.AsEnumerable().Reverse().ToArray();
         }
+
+        private static string BestRoot(string[][] pathParts)
+        {
+            var roots = pathParts.Select(x => x[0])
+                .GroupBy(x => x)
+                .Select(x => new { Root = x.Key, Count = x.Count() })
+                .OrderByDescending(x => x.Count)
+                .ToArray();
+            if (roots.Length == 0)
```

Comment:
this should be at the beginning? because it can be 0 only if `pathParts` is empty?

given this is private static, maybe even not invoke this if the `pathParts` is empty, to avoid returning `null`

Reply:
I tried, but it leads to an ugly code. Check [this intermediate commit](https://github.com/SonarSource/sonar-scanner-msbuild/pull/1307/commits/be3879804aa23d66978cb2fffc1fcf2fdc2476da) that lead to coverage issues due to state overlaps (one logic spread over multiple places)

I had `!paths.Any()` as an early bailout. The problem with `BestRoot` is then that it will test `1` and `>1` and leads to a consistency gap `what about 0?`.

The current implementation visits some paths that it doesn't have to, but:
* proves consistency
* should not happen rarely
* is hit once per analysis
* doing Select/GroupBy/Select/Order/ToArray on empty collection is cheap